Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -298,42 +306,186 @@ public class ProgramBuilder {
+
+        // The variable that we will return.
+        var obj: Variable
+
+        // Fast path for array creation.
+        if type.Is(fuzzer.environment.arrayType) && probability(0.9) {
+            print("took fast path")
+            let value = withEqualProbability({
+                self.loadFloat(13.37)
+            }, {
+                self.loadInt(1337)
+            }, {
+                self.createObject(with: [:])
+            })
+            return self.createArray(with: Array(repeating: value, count: Int.random(in: 1...5)))
+        }
+
+        if let group = type.group {
+            // We check this during Environment initialization, but let's keep this just in case.
+            assert(self.fuzzer.environment.type(ofBuiltin: group) != .unknown, "We don't know how to construct \(group)")
+            let constructionSignature = self.fuzzer.environment.type(ofBuiltin: group).constructorSignature!
+            let arguments = self.generateCallArguments(for: constructionSignature)
+            let constructor = self.loadBuiltin(group)
+            obj  = self.construct(constructor, withArgs: arguments)
+        } else {
+            // Either generate a literal or use the store property stuff.
+            if probability(0.8) { // Do the literal
+                var initialProperties: [String: Variable] = [:]
+                // gather properties of the correct types
+                for prop in type.properties {
+                    let type = interpreter?.type(ofProperty: prop)
+                    var value: Variable?
+                    if type != nil && type != .unknown {
+                        value = self.randVar(ofConservativeType: type!)
+                        if value == nil {
+                            value = generateVariable(ofType: type!)
+                        }
+                    } else {
+                        value = self.randVar()
+                    }
+                    initialProperties[prop] = value
+                }
+                // TODO: This should take the method type/signature into account!
+                _ = type.methods.map { initialProperties[$0] = self.randVar(ofType: .function())! }
+                obj = self.createObject(with: initialProperties)
+            } else { // Do it with storeProperty
+                obj = self.createObject(with: [:])
+                for method in type.methods {
+                    // TODO: This should take the method type/signature into account!
+                    let methodVar = self.randVar(ofType: .function())
+                    self.storeProperty(methodVar!, as: method, on: obj)
+                }
+                // These types might have been defined in the interpreter
+                for prop in type.properties {
+                    let type = interpreter?.type(ofProperty: prop)
+                    var value: Variable?
+                    if type != nil && type != .unknown {
```

Comment:
Nit: I think you can write this as `if let type = interpreter?.type(ofProperty: prop), type != .unknown {`

Reply:
Actually, can't you use `ProgramBuilder.type(ofProperty:)` here?

Reply:
ah cool, was looking for something like that! :)
yeah I added the `ProgramBuilder.type(ofProperty:)` function after changing this I think?